test/unit/GoogleAnalyticsEventProcessor/processEvent_test.js

```diff
@@ -0,0 +1,319 @@
+      eventProcessor.processEvent(
+        emptyStorage,
+        emptyModel,
+        'test',
+        {});
+
+      expect(jasmine.Ajax.requests.mostRecent().method).toBe('POST');
+    });
+
+    it('has the request url built by the event processor', () => {
+      eventProcessor.processEvent(
+        emptyStorage,
+        emptyModel,
+        'test',
+        {});
+
+      expect(jasmine.Ajax.requests.mostRecent().url).toBe('mock_request_url');
+    });
+
+    it('includes the client ID retrieved by event processor', () => {
+      eventProcessor.processEvent(
+        emptyStorage,
+        emptyModel,
+        'test',
+        {});
+
+      expect(jasmine.Ajax.requests.mostRecent().data().clientId)
+          .toBe('mock_client_id');
+    });
+
+    it('creates a single event within the events array', () => {
+      eventProcessor.processEvent(
+        emptyStorage,
+        emptyModel,
+        'test',
+        {});
+
+      expect(jasmine.Ajax.requests.mostRecent().data().events.length)
+          .toBe(1);
+    });
+
+    it('adds the event name to the single event', () => {
+      eventProcessor.processEvent(
+        emptyStorage,
+        emptyModel,
+        'test',
+        {});
+
+      expect(jasmine.Ajax.requests.mostRecent().data().events[0].name)
+          .toBe('test');
+    });
+
+    it('has no event parameters when both model and eventOptions ' +
+        'are empty', () => {
+      eventProcessor.processEvent(
+        emptyStorage,
+        emptyModel,
+        'test',
+        {});
+
+      expect(jasmine.Ajax.requests.mostRecent().data().events[0].params)
+          .toEqual({});
+    });
+
+    it('has all event parameters passed in via eventOptions ' +
+        'that are defined', () => {
+      const eventOptions = {
+        param_one: 1,
+        param_two: '2',
+        param_three: null,
+        param_four: {
+          param_five: true,
+        },
+        param_six: ['test'],
+        param_seven: false,
+      };
+
+      eventProcessor.processEvent(
+        emptyStorage,
+        emptyModel,
+        'test',
+        eventOptions);
+
+      expect(jasmine.Ajax.requests.mostRecent().data().events[0].params)
+          .toEqual(eventOptions);
+    });
+
+    it('has event level global values when key is defined in model and ' +
+        'is included in default automatic parameter list', () => {
+      eventProcessor.processEvent(
+        emptyStorage,
+        new MockModelInterface({
+          page_path: 'model_page_path',
+        }),
+        'test',
+        {});
+
+      expect(
+        jasmine.Ajax.requests.mostRecent().data().events[0].params.page_path
+      ).toBe('model_page_path');
+    });
+
+    it('has event level global values when key is defined in model and ' +
+        'is added to automatic parameter list', () => {
+      eventProcessor.processEvent(
+        emptyStorage,
+        new MockModelInterface({
+          added_param: 'model_added_param',
```

Comment:
I would add a "not_added_param" to the global model, either here or in a separate test, and an assert that it doesn't get added.